samples/openapi3/client/petstore/python/test/test_mole.py

```diff
@@ -0,0 +1,50 @@
+"""
+    OpenAPI Petstore
+
+    This spec is mainly for testing Petstore server and contains fake endpoints, models. Please do not use this for any other purpose. Special characters: \" \\  # noqa: E501
+
+    The version of the OpenAPI document: 1.0.0
+    Generated by: https://openapi-generator.tech
+"""
+
+
+import sys
+import unittest
+
+import petstore_api
+from petstore_api.model.mole import Mole
+
+
+class TestMole(unittest.TestCase):
+    """Mole unit test stubs"""
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testMole(self):
+        """Test Mole"""
+        mole = Mole(smell="dirt")
+        assert mole.smell == "dirt"
+
+        mole = Mole(smell="dirt", taste="kfc")
+        assert mole.smell == "dirt"
+        assert mole.taste == "kfc"
+
+        with self.assertRaises(Exception):
+            Mole(smell="dirt", blind=True)
+
+        with self.assertRaises(Exception):
+            Mole(smell="dirt", touch=True)
+
+        mole = Mole.from_openapi_data(smell="dirt", taste="kfc", blind=True, touch=False)
+        assert mole.smell == "dirt"
+        assert mole.taste == "kfc"
+        assert mole.blind is True
+        assert mole.touch is False
```

Comment:
These tests look great, thank you for adding them!
A couple of points.
- Can you move it to https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/petstore/python/tests_manual? That's where we store manually written tests. We like to have the tests in tests be the autogen tests with no human written content
- can you add comments describing what each instantiation is testing?
```
# includes required parameters that are not readOnly
mole = Mole(smell="dirt")

# includes required parameters that are not readOnly and an optional readOnly false arg
mole = Mole(smell="dirt", taste="kfc")

# passing in required readOnly parameters raises an exception
with self.assertRaises(Exception):
    Mole(smell="dirt", blind=True

# passing in optional readOnly parameters raises an exception
with self.assertRaises(Exception):
    Mole(smell="dirt", touch=True)

# passing in required an optional parameters with readOnly true or false works with from_openapi_data
mole = Mole.from_openapi_data(smell="dirt", taste="kfc", blind=True, touch=False)
```

For your assertRaises, can you check for specific  ApiX errors ApiAttributeError etc? It helps to cast all exceptions into our defined exceptions because then our users can capture all OpenApiExceptions or ApiAttributeError or AttributeError because all of those are included in the ApiAttributeError class.

Reply:
I thought the test in https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/petstore/python/test where suppose to the Model class. The auto-generated tests are empty and therefore meaningless. Are you planing to generate tests that are at least a bit useful?

 https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/petstore/python/tests_manual
 was more to test the endpoints.
 
 Anyway, I can do that if there is a plan for auto-generated tests

Reply:
We have autogen tests there for models and apis. They are a scaffold that can be built upon.
Long term we should improve those tests and auto-generate some positive and negative test cases.
One runs into the issue that for composed schemas, one does not know which oneOf, anyOf, allOf schemas to select when making an example payload, so generating a payload to test with is non-trivial.
For primitive models and object models it's easy and doable.

Some effort has been invested in generating some model tests in the python-legacy client, as seen [here](ttps://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python-legacy/test/test_pet.py)
But that effort results in broken autogen python composed schema tests because our java code doesn't know how to generate composed schema tests, or values for properties that are of type composed schema.

Reply:
For a composed schema one can have complicated use cases like this:
```
OneOfSchema:
  oneOf:
    - type: object
      properties:
        a:
          type: integer
    - type: object
      properties:
        b:
          type: integer
    - type: object
      properties:
        c:
          type: integer

ComplicatedComposed:
  properties:
    id:
      type: integer
  allOf:
    - OneOfSchema
  oneOf:
    - type: object
      properties:
        d:
          type: integer
    - type: object
      properties:
        e:
          type: integer
    - type: object
      properties:
        d:
          type: integer
```
What should go in the payload for ComplicatedComposed?
We need one of a,b,c and id, and one of d, e f
So a, id, d is valid, but it must also validate for each of the schemas:
- OneOfSchema
- the chosen oneOf schema in OneOfSchema
- ComplicatedComposed
- the chosen oneOf schema in ComplicatedComposed

And this example doesn't even cover :
- cases where variables have the same name but different types in schemas or
- parameter may be different types
- payloads not meeting formatting or validation constraints